File: examples/CSharp/SimpleExample.cs
using System;
using System.Net.Http;
using NBomber.CSharp;
using NBomber.Http;
using NBomber.Http.CSharp;
using NBomber.Plugins.Network.Ping;

namespace CSharp;

class SimpleExample
{
    public void Run()
    {
        // For this example, you'll need to start the HttpApiSimulator, which is located in the examples/simulators solution folder.
        // Make sure it’s running before executing the client tests to ensure proper communication.

        using var httpClient = Http.CreateDefaultClient();
        var url = "http://localhost:5071/api/pingpong";

        var scenario = Scenario.Create("http_scenario", async context =>
        {
            var request =
                Http.CreateRequest("GET", url)
                    .WithHeader("Accept", "application/json")
                    .WithBody(new StringContent("{ some JSON }"));

            var response = await Http.Send(httpClient, request);

            return response;
        })
        .WithWarmUpDuration(TimeSpan.FromSeconds(3))
        .WithLoadSimulations(Simulation.Inject(rate: 1000, interval: TimeSpan.FromSeconds(1), during: TimeSpan.FromSeconds(30)));

        NBomberRunner
            .RegisterScenarios(scenario)
            .WithWorkerPlugins(
                new HttpMetricsPlugin([HttpVersion.Version1])
            )
            .Run();
    }
}

File: examples/FSharp/SimpleExample.fs
module SimpleExample

open System.Net.Http
open NBomber
open NBomber.Contracts
open NBomber.Http
open NBomber.Http.FSharp
open NBomber.FSharp
open NBomber.Plugins.Network.Ping

let run () =

    // For this example, you'll need to start the HttpApiSimulator, which is located in the examples/simulators solution folder.
    // Make sure it’s running before executing the client tests to ensure proper communication.

    use httpClient = Http.createDefaultClient()
    let url = "http://localhost:5071/api/pingpong"

    Scenario.create("http_scenario", fun context -> task {

        let! response =
            Http.createRequest "GET" url
            |> Http.withHeader "Accept" "application/json"
            |> Http.withBody (new StringContent("{ some JSON }"))
            |> Http.send httpClient

        // let user = {| Id = "1"; Name = "Test Name" |}
        //
        // let! response =
        //     Http.createRequest "GET" "https://nbomber.com"
        //     |> Http.withJsonBody user
        //     |> Http.send httpClient

        return response
    })
    |> Scenario.withWarmUpDuration(seconds 3)
    |> Scenario.withLoadSimulations [Inject(rate = 1000, interval = seconds 1, during = minutes 1)]
    |> NBomberRunner.registerScenario
    |> NBomberRunner.withWorkerPlugins [
        new PingPlugin(PingPluginConfig.createDefault "nbomber.com")
        new HttpMetricsPlugin()
    ]
    |> NBomberRunner.run
    |> ignore
